test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	componentos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/assert"
+)
+
+// generateLog generates and verifies log contents.
+func generateLog(s *LinuxVMFakeintakeSuite, content string) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+	t := s.T()
+
+	// Get OS type using the OSType from vm_client.go from e2e/client package
+	osType := s.Env().VM.GetOSType()
+	var os string
+
+	switch osType {
+	case componentos.WindowsType:
+		os = "windows"
+		t.Log("Generating Windows log.")
+		logPath = "C:\\logs\\hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("Get-Content %s", logPath)
+	default: // Assuming Linux if not Windows.
+		os = "linux"
+		t.Log("Generating Linux log.")
+		logPath = "/var/log/hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("cat %s", logPath)
+	}
+
+	s.Env().VM.Execute(cmd)
+
+	// Check if the log has been generated.
+	output := s.Env().VM.Execute(checkCmd)
+	if strings.Contains(output, content) {
+		t.Logf("Finished generating %s log.", os)
+	}
+}
+
+// checkLogs checks and verifies logs inside the intake.
+func checkLogs(suite *LinuxVMFakeintakeSuite, service, content string) {
+	client := suite.Env().Fakeintake
+
+	suite.EventuallyWithT(func(c *assert.CollectT) {
+		names, err := client.GetLogServiceNames()
+		if !assert.NoErrorf(c, err, "Error found: %s", err) {
+			return
+		}
+
+		if len(names) > 0 {
+			logs, err := client.FilterLogs(service)
+			if !assert.NoErrorf(c, err, "Error found: %s", err) {
+				return
+			}
+			if !assert.NotEmpty(c, logs, "No logs with service matching '%s' found, instead got '%s'", service, names) {
+				return
+			}
+
+			logs, err = client.FilterLogs(service, fi.WithMessageContaining(content))
+			assert.NoErrorf(c, err, "Error found: %s", err)
+			assert.NotEmpty(c, logs, "Expected at least 1 log with content: '%s', but received %v logs.", content, logs)
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+}
+
+// cleanUp cleans up any existing log files (only useful when running dev mode/local runs).
```

Comment:
why is this only useful in dev mode? It looks like we currently run this after every test, should we only run it after every test if `devMode == true`?

Reply:
Overall since the log files will get removed outside of devmode, ill adjust the cleanup script so that it will only remove the files if dev mode is enabled. As for the intake, i think its necessary for the intake to be cleaned up after every test